source/aws_signing.c

```diff
@@ -802,6 +830,57 @@ static int s_validate_query_params(struct aws_array_list *params) {
     return AWS_OP_SUCCESS;
 }
 
+/**
+ * Apply or remove URI-encoding to each aws_uri_param in a list.
+ * (new strings are added to temp_strings)
+ */
+static int s_transform_query_params(
+    struct aws_signing_state_aws *state,
+    struct aws_array_list *param_list,
+    struct aws_array_list *temp_strings,
+    int (*byte_buf_append_param_fn)(struct aws_byte_buf *, const struct aws_byte_cursor *)) {
```

Comment:
Might add a note that the append function should be dynamic and not fail if there's not enough room